### ISSUE TITLE
OpenSwathWorkflow test failure

### ISSUE THREAD

**Issue by hroest** [OpenMS/OpenMS#1163]
OpenSwathWorkflow seems to fail under MSVS 2012
```
TOPP_OpenSwathWorkflow_2_out1 	Failed 	900ms 	Completed (Failed)
TOPP_OpenSwathWorkflow_1_out1 	Failed 	840ms 	Completed (Failed)
TOPP_OpenSwathWorkflow_3_out1 	Failed 	810ms 	Completed (Failed)
```

and at least one issue is the Signal to Noise score:
```

in1:  C:\dev\NIGHTLY\builds\win8-VS12_x86-Debug\src\tests\topp\OpenSwathWorkflow_1.featureXML.tmp   (line: 82, position/column: 50/72)
			<userParam type="float" name="sn_ratio" value="!
			                                               2.12550163726256"/>

in2:  C:\dev\NIGHTLY\OpenMS\src\tests\topp\OpenSwathWorkflow_1_output.featureXML   (line: 82, position/column: 50/72)
			<userParam type="float" name="sn_ratio" value="!
			                                               2.06010679495179"/>


Easy Access:
C:\dev\NIGHTLY\builds\win8-VS12_x86-Debug\src\tests\topp\OpenSwathWorkflow_1.featureXML.tmp:82:72:
```

**Comment by hroest** [comment]
no issue on Mac, Ubuntu Linux 12.04 or Windows MSVS 2010 so far

**Comment by hroest** [comment]
SN Score is calculated here

https://github.com/OpenMS/OpenMS/blob/develop/src/openswathalgo/source/ANALYSIS/OPENSWATH/OPENSWATHALGO/ALGO/MRMScoring.cpp#L324

the `ISignalToNoisePtr` is created here https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp#L245

and the function that is called is found here https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/ANALYSIS/OPENSWATH/DATAACCESS/MRMFeatureAccessOpenMS.h#L175

**Comment by timosachsenberg** [comment]
no issue on Win 8.1, VS2013 SP1

**Comment by hroest** [comment]
very strange, I see the issue also under 14.04 now. While debugging I changed the following in the MRMFeatureAccess class of OpenMS in order to hunt down the bug:

```
--- a/src/openms/include/OpenMS/ANALYSIS/OPENSWATH/DATAACCESS/MRMFeatureAccessOpenMS.h
+++ b/src/openms/include/OpenMS/ANALYSIS/OPENSWATH/DATAACCESS/MRMFeatureAccessOpenMS.h
@@ -175,7 +175,11 @@ public:
     double getValueAtRT(double RT)
     {
       typename OpenMS::MSSpectrum<PeakT>::const_iterator it = chromatogram_.MZBegin(RT);
-      return sn_.getSignalToNoise(*it);
+      std::cout <<" get iterator here " << *it << " at " << it->getMZ() << " --> for RT " << RT << std::endl;
+      double ret = sn_.getSignalToNoise(*it);
+      std::cout << "   return " << ret << ";" << std::endl;
+      return ret;
     }

```


under Ubuntu 12.04 I get

```
1364: Will analyze 18 transitions in total.
1364: Progress of 'Extracting and scoring transitions':
1364: Thread 0 will analyze 1 peptides and 3 transitions from SWATH 1 in batches of 1
1364: Warning: MRMFeatureFinderScoring received the unknown parameter 'EMGScoring:deltaRelError'!
1364:  get iterator here POS: 111 INT: 1392.01 at 111 --> for RT 101
1364:    return 2.06011;
1364:  get iterator here POS: 111 INT: 696.003 at 111 --> for RT 101
1364:    return 2.06011;
1364:  get iterator here POS: 111 INT: 464.002 at 111 --> for RT 101
1364:    return 2.06011;
[...]
```

under Ubuntu 14.04 I get

```
1364: Will analyze 18 transitions in total.
1364: Progress of 'Extracting and scoring transitions':
1364: Thread 0 will analyze 1 peptides and 3 transitions from SWATH 1 in batches of 1
1364: Warning: MRMFeatureFinderScoring received the unknown parameter 'EMGScoring:deltaRelError'!
1364:  get iterator here POS: 101 INT: 1436.19 at 101 --> for RT 101
1364:    return 2.1255;
1364:  get iterator here POS: 101 INT: 718.096 at 101 --> for RT 101
1364:    return 2.1255;
1364:  get iterator here POS: 101 INT: 478.731 at 101 --> for RT 101
1364:    return 2.1255;
```

**Comment by hroest** [comment]
so it seems to be an issue with `OpenMS::MSSpectrum<PeakT>::MZBegin(RT)` which does not return the same value when given the *exact* same value as the peak. It seems that under Ubuntu 12.04 it does return a value that is larger but in 14.04 it does return a value that is equal. According to the standard http://en.cppreference.com/w/cpp/algorithm/lower_bound
```
Returns an iterator pointing to the first element in the range [first, last) that is not less than (i.e. greater or equal to) value. 
```
so it seems the behavior in Ubuntu 14.04 is actually correct

**Comment by cbielow** [comment]
If I read this correctly, VS2012 and Ubuntu14.04 failed the tests, but were actually correct when computing lower_bound, whereas all the other compilers got it wrong?!

I find it hard to believe that something so elemental is broken. Sounds more like a type conversion issue (float/double) or some other side effect, but I cant find it on first sight...

**Comment by hroest** [comment]
can someone confirm that #1189  also works on VS2012 ?

**Comment by hroest** [comment]
closed through #1189
